File: Oauth2Simples/src/main/java/br/com/estudos/oauth2/repository/UsuarioRepository.java
package br.com.estudos.oauth2.repository;

import java.util.Optional;

import br.com.estudos.oauth2.model.Usuario;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;


@Repository
public interface UsuarioRepository extends CrudRepository<Usuario,Long>{

     Optional<Usuario> findByEmail(String email);

}

File: kotlin/kotlinEstudos/src/main/java/br/com/kotlin/estudos/repository/UsuarioRepository.kt
package br.com.kotlin.estudos.repository

import br.com.kotlin.estudos.model.repository.Usuario

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UsuarioRepository : CrudRepository<Usuario, Long> {

    fun findByEmail(email: String): Optional<Usuario>

}